apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporterTest.java

```diff
@@ -0,0 +1,214 @@
+import com.apollocurrency.aplwallet.apl.core.db.BlockDaoImpl;
+import com.apollocurrency.aplwallet.apl.core.db.DatabaseManager;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.DerivedDbTablesRegistryImpl;
+import com.apollocurrency.aplwallet.apl.core.db.ShardDaoJdbcImpl;
+import com.apollocurrency.aplwallet.apl.core.db.TransactionalDataSource;
+import com.apollocurrency.aplwallet.apl.core.db.cdi.transaction.JdbiHandleFactory;
+import com.apollocurrency.aplwallet.apl.core.db.dao.BlockIndexDao;
+import com.apollocurrency.aplwallet.apl.core.db.dao.TransactionIndexDao;
+import com.apollocurrency.aplwallet.apl.core.db.derived.DerivedTableData;
+import com.apollocurrency.aplwallet.apl.core.db.fulltext.FullTextConfigImpl;
+import com.apollocurrency.aplwallet.apl.core.db.fulltext.FullTextSearchEngine;
+import com.apollocurrency.aplwallet.apl.core.db.fulltext.FullTextSearchService;
+import com.apollocurrency.aplwallet.apl.crypto.Convert;
+import com.apollocurrency.aplwallet.apl.data.BalancesPublicKeysTestData;
+import com.apollocurrency.aplwallet.apl.data.DbTestData;
+import com.apollocurrency.aplwallet.apl.extension.DbExtension;
+import com.apollocurrency.aplwallet.apl.extension.TemporaryFolderExtension;
+import com.apollocurrency.aplwallet.apl.util.env.config.Chain;
+import com.apollocurrency.aplwallet.apl.util.env.dirprovider.ConfigDirProvider;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import lombok.extern.slf4j.Slf4j;
+import org.jboss.weld.junit.MockBean;
+import org.jboss.weld.junit5.EnableWeld;
+import org.jboss.weld.junit5.WeldInitiator;
+import org.jboss.weld.junit5.WeldSetup;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+
+@Slf4j
+@EnableWeld
+class GenesisImporterTest {
+
+    @RegisterExtension
+    DbExtension extension = new DbExtension(DbTestData.getDbFileProperties(createPath("genesisImport").toAbsolutePath().toString()));
+    @RegisterExtension
+    static TemporaryFolderExtension temporaryFolderExtension = new TemporaryFolderExtension();
+
+    private BlockchainConfig blockchainConfig = mock(BlockchainConfig.class);
+    private BlockchainConfigUpdater blockchainConfigUpdater = mock(BlockchainConfigUpdater.class);
+    private HeightConfig config = Mockito.mock(HeightConfig.class);
+    private Chain chain = Mockito.mock(Chain.class);
+    private ConfigDirProvider configDirProvider = mock(ConfigDirProvider.class);
+    private AplAppStatus aplAppStatus = mock(AplAppStatus.class);
+
+    @WeldSetup
+    public WeldInitiator weld  = WeldInitiator.from(
+            AccountTable.class, FullTextConfigImpl.class, DerivedDbTablesRegistryImpl.class, PropertiesHolder.class,
+            ShardDaoJdbcImpl.class,
+            BlockIndexDao.class, TransactionDaoImpl.class, BlockchainImpl.class,
```

Comment:
Do you really require jdbi?

Reply:
Yes, it's needed by dependent components here.